lib/checkPullRequestJob.js

```diff
@@ -0,0 +1,266 @@
+      definition.indexOf('(')
+    );
+  });
+
+  return newJobs;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const addsNewJob = (file) => {
+  const newJobs = getNewJobsFromFile(file);
+  return newJobs.length > 0;
+};
+
+/**
+ * Returns the name of the job from it's file name.
+ * @param {string} filename
+ *
+ * @example
+ * getJobNameFromFileName('core/domain/user_jobs_one_off.py')
+ * // returns user_jobs_one_off
+ */
+const getJobNameFromFileName = (filename) => {
+  // Filename is the relative path to the file, we need to
+  // get the exact file name.
+  const substrings = filename.split('/');
+  const actualFileNameWithExtension = substrings[substrings.length - 1];
+  const extensionIndex = actualFileNameWithExtension.indexOf('.py');
+
+  return actualFileNameWithExtension.substring(0, extensionIndex);
+};
+
+/**
+ * @param {string} job
+ * @param {import('probot').Octokit.PullsListFilesResponse} files
+ */
+const isRegistryUpdated = (job, files) => {
+  const registryFile = files.find(({filename}) => {
+    return filename.includes(REGISTRY_FILENAME);
+  });
+
+  return registryFile && registryFile.patch.includes(job);
+};
+
+/**
+ * Gets all the changed files in the repo
+ * @param {import('probot').Context} context
+ * @returns {Promise<import('probot').Octokit.PullsListFilesResponseItem[]>}
+ */
+const getAllChangedFiles = async (context) => {
+  const changedFiles = [];
+  /**
+   * @type {import('probot').Octokit.PullsGetResponse}
+   */
+  const pullRequest = context.payload.pull_request;
+  // 100 is used here since that is the maximum files per
+  // request that can be obtained from the API.
+  const maxFilesPerPage = 100;
+
+  let totalFilesFetched = 0;
+  let currentPage = 1;
+  do {
+    const fileListParams = context.repo({
+      pull_number: pullRequest.number,
+      per_page: maxFilesPerPage,
+      page: currentPage,
+    });
+
+    const fileResponse = await context.github.pulls.listFiles(fileListParams);
+    changedFiles.push(...fileResponse.data);
+    totalFilesFetched += fileResponse.data.length;
+    currentPage++;
+  } while (totalFilesFetched < pullRequest.changed_files);
+
+  return changedFiles;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} newJobFiles
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ */
+const getRegistryReminder = (newJobFiles, changedFiles) => {
+  let registryReminder = ', ';
+
+  const hasNotAddedFileToRegistry = newJobFiles.some(
+    (file) => {
+      const jobs = getNewJobsFromFile(file);
+      const jobName = getJobNameFromFileName(file.filename);
+      return jobs.some((job) => !isRegistryUpdated(
+        `${jobName}.${job}`, changedFiles ));
+    }
+  );
+
+  if (hasNotAddedFileToRegistry) {
+    const fileText = newJobFiles.length > 1 ? 'files' : 'file';
+    registryReminder +=
+      'please add the new job ' + fileText + ' to the job registry and ';
```

Comment:
to the job registry [Link to jobs_registry.py]

Reply:
Done.